portal-web/docroot/html/taglib/ui/search_iterator/init.jsp

```diff
@@ -38,4 +37,17 @@ List resultRows = searchContainer.getResultRows();
 String summary = searchContainer.getSummary();
 
 JSONArray primaryKeysJSONArray = JSONFactoryUtil.createJSONArray();
+%>
+
+<%!
+private String _getRowElementId(String partialId, PortletRequest portletRequest, PortletResponse portletResponse, com.liferay.portal.kernel.dao.search.ResultRow row) {
+	String rowElementId = AUIUtil.getNamespace(portletRequest, portletResponse)
+		.concat(partialId.concat(StringPool.UNDERLINE.concat(row.getRowId())));
+
+	if (row.getPrimaryKey() != null) {
+		rowElementId = rowElementId.concat(StringPool.UNDERLINE.concat(row.getPrimaryKey()));
+	}
+
+	return rowElementId;
+}
```

Comment:
1. Ah you're right, namespace will always be there. My bad. :+1: 
2. I think there is no need to import, but I'm not 100% sure. This code is in the only `portal-web`, not in a typical portlet, and there are some things that behave differently here.
3. Looks mostly good, I'd only put a minor change, something like
```java
StringBundler sb = new StringBundler(row.getPrimaryKey() == null ? 4 : 6);
```

Reply:
2. I guess I'm missing something. I'm trying to use `liferayPortletResponse.getNamespace()` but it doesn't work when I deploy the code. I saw other files but they look like it just uses it, no need to import. (as you said) That probably doesn't work inside portal-web. I'll keep the `AUIUtil`.
3. Great, I'll change that.

Reply:
Looking for `getNamespace` in `portal-web` , I see
- `portletDisplay.getNamespace()`
- `PortalUtil.getLiferayPortletResponse(liferayRenderResponse)).getNamespace()`
- liferayPortletResponse.getNamespace()
- renderResponse.getNamespace()

Does any of these work? :sweat_smile: 

I'm trying to avoid any use if AUI, if possible, because we eventually want to remove it all from DXP.

Reply:
`portletResponse.getNamespace()` works fine.

I see, I did not know we're going to remove AUI from DXP.

Reply:
Changes applied, resolving this conversation.